### PR TITLE
Update to zig 0.11.0

### PR DESCRIPTION
**Pull request by psyomn**
Hi there, hope you're doing well!

I was fiddling around with your library, and decided to give it a shot at updating to the latest release.  I'm not sure if any of these patches are welcome, but after my changes started piling up, I decided to post them here. 

I think the only thing that remains to get working is the `zig build test` step, but once that's done, the PR should technically be ready.

Cheers!

### CODE REVIEW

**Comment by ikskuh** [comment]
Very nice, thanks for the work! I very welcome updates to my libs!

File: src/lib.zig [hunk 1]
```diff
@@ -246,7 +246,10 @@ export fn gemtextDocumentRemove(document: *c.gemtext_document, index: usize) voi
         );
     }
 
-    fragments = allocator.shrink(fragments, fragments.len - 1);
+    fragments = allocator.realloc(fragments, fragments.len - 1) catch |err| t: {
+        std.log.warn("could not resize fragments: {}", .{err});
+        break :t fragments;
+    };
```
Comment (psyomn) [inline src/lib.zig:252]: So I took a look at the allocators, but I'm getting the impression that `shrink` may be no more.  There's `resize`, but when I tried using it here, the slice len was not updated, which caused bad things :tm: 

If you have a better suggestion I'm more than happy to change this to something better.

**Comment by psyomn** [comment]
AFAICT, things seem to be working properly: 
```
[psyomn@xaos zig-gemtext]$ zig build test examples 
[psyomn@xaos zig-gemtext]$ echo $?
0

[psyomn@xaos bin]$ for cmd in gem2html-zig gem2md-zig gem2html-c gem2md-c; do echo "$cmd"; echo -e "# hello\nthere" | ./$cmd; echo $?; done
gem2html-zig
<h1>hello</h1>
<p>there</p>
0
gem2md-zig
# hello

there
0
gem2html-c
<h1>hello</h1>
<p>there</p>
0
gem2md-c
# hello
there
0
```

Happy to change things, be it from code to commit messages.  Let me know!

File: src/lib.zig (hunk 1 above)
Comment (ikskuh) [reply src/lib.zig:252]: If resize fails, you can do a malloc and free to create a new slice and basically implement `realloc` like it behaves in C

Comment (psyomn) [reply src/lib.zig:252]: So the odd thing is that my original use was with `resize`, and I did indeed check the return for a fail.  The return of `resize` returned `true`, however the slice len remained `3`.  The discriminator of the _last_ fragment after the `memcpy` seemed corrupt, too, which is what I was getting at with the bad things :tm:. 

Let me know what you would prefer, I can give a shot at what you're talking about too!

**Comment by ikskuh** [comment]
Ah you know what? I'm fine with the code!

**Comment by psyomn** [comment]
Hey by the way, I hope I didn't rush you into merging this -- that was
not my intent.

I'm not sure I fully understood what you meant with the resize comment
-- did you mean that an approach could be to first attempt `resize`, and
on fail, then only try to do a realloc in order to save an alloc?

I'm also looking at the allocator code, but from what I understand it
[1] takes into consideration these optimization cases (though I might be
wrong, as I'm still learning the language).

[1] https://github.com/ziglang/zig/blob/master/lib/std/mem/Allocator.zig#L255

Anyway, I hope I didn't cause too much trouble; just trying to get a
little better at the language!

**Pull request merged**